src/contracts/ParaSwapDebtSwapAdapter.sol

```diff
@@ -141,9 +176,34 @@ abstract contract ParaSwapDebtSwapAdapter is
     require(msg.sender == address(POOL), 'CALLER_MUST_BE_POOL');
     require(initiator == address(this), 'INITIATOR_MUST_BE_THIS');
 
-    FlashParams memory swapParams = abi.decode(params, (FlashParams));
-    _swapAndRepay(swapParams, IERC20Detailed(assets[0]), amounts[0]);
+    FlashParams memory flashParams = abi.decode(params, (FlashParams));
+
+    // This is only non-zero if we flashed extra collateral.
+    if (flashParams.nestedFlashloanDebtAsset != address(0)) {
+      // Wrap the swap with a supply and withdraw.
+      address collateralAsset = assets[0];
+      uint256 collateralAmount = amounts[0];
+
+      // Supply
+      _supply(collateralAsset, collateralAmount, flashParams.user, REFERRER);
 
+      // Execute the nested flashloan
+      address newAsset = flashParams.nestedFlashloanDebtAsset;
```

Comment:
`newDebtAsset`?

Reply:
Can't be because in this scenario it's not the debt asset but the collateral, it's just the asset we flash